syntax = "proto3";

package filetransfer;

message UploadFileInfo{
    int32 code = 1;
    string data = 2;
}

message UploadByte{
    int32 code = 1;
    string data = 2;
}

message FinishFile{
    int32 code = 1;
}

message ResponseStatus{
    int32 code = 1;
}

service FileTransfer{
    rpc SetFileInfo (UploadFileInfo) returns (ResponseStatus) {}
    rpc UploadFile (UploadByte) returns (ResponseStatus) {}
    rpc EndFileStream (FinishFile) returns (ResponseStatus) {}
}